### PR TITLE
feat(network_info_plus)!: Remove deprecated permission handling methods, update example and docs

### PR DESCRIPTION
**Pull request by miquelbeltran**
## Description

- Removed the deprecated permission handling methods. Instead, devs should use the permission_handler plugin.
- Updated example code with the permission_handler code.
  - Note that this also requires modifying the project Podfile, but this file is ignored by the repo.
- Updated example iOS project configuration.
  - Added location permission to the `info.plist`.
  - Added wifi entitlement to the `Runner.entitlements`.
  - General iOS project boilerplate changes.
- Update README documentation.

Note: I couldn't get this working because I don't have a professional apple development account anymore. This is necessary to have the Wi-Fi entitlement permission. I am not really wanting to pay 100 USD just to test this 😩 

## Related Issues

- Permission issue discussed in https://github.com/fluttercommunity/plus_plugins/issues/2156

## Checklist

- [x] I read the [Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the plugin version in `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [ ] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/network_info_plus/network_info_plus/example/android/app/build.gradle [hunk 1]
```diff
@@ -26,7 +26,7 @@ apply plugin: 'kotlin-android'
 apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
 
 android {
-    compileSdkVersion 33
+    compileSdkVersion 34
```
Comment (miquelbeltran) [inline packages/network_info_plus/network_info_plus/example/android/app/build.gradle:29]: had to increase the min sdk of the example for running on Android

File: packages/network_info_plus/network_info_plus/example/ios/Runner/Info.plist [hunk 2]
```diff
@@ -24,6 +26,10 @@
 	<string>$(FLUTTER_BUILD_NUMBER)</string>
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string>Obtain WIFI Name</string>
```
Comment (miquelbeltran) [inline packages/network_info_plus/network_info_plus/example/ios/Runner/Info.plist:30]: Added the location permission to the info.plist as recommended in the plugin readme

File: packages/network_info_plus/network_info_plus/example/ios/Runner/Runner.entitlements [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>com.apple.developer.networking.wifi-info</key>
+	<true/>
+</dict>
+</plist>
```
Comment (miquelbeltran) [inline packages/network_info_plus/network_info_plus/example/ios/Runner/Runner.entitlements:8]: Added the necessary entitlement. As documented, only people who pay in the apple developer portal can develop apps with this.

File: packages/network_info_plus/network_info_plus/example/lib/main.dart [hunk 4]
```diff
@@ -97,18 +98,13 @@ class _MyHomePageState extends State<MyHomePage> {
...
-        if (status == LocationAuthorizationStatus.notDetermined) {
-          // ignore: deprecated_member_use
-          status = await _networkInfo.requestLocationServiceAuthorization();
-        }
-        if (status == LocationAuthorizationStatus.authorizedAlways ||
-            status == LocationAuthorizationStatus.authorizedWhenInUse) {
+      if (!kIsWeb && (Platform.isAndroid || Platform.isIOS)) {
+        // Request permissions as recommended by the plugin documentation:
+        // https://github.com/fluttercommunity/plus_plugins/tree/main/packages/network_info_plus/network_info_plus
+        if (await Permission.locationWhenInUse.request().isGranted) {
```
Comment (miquelbeltran) [inline packages/network_info_plus/network_info_plus/example/lib/main.dart:104]: "location when in use" should be enough for this example. If a user wants to use "location always" the code gets a bit more complicated, according to the permission_handler plugin devs. (we were probably doing that wrong in the plugin)

**Review by vbuberen — APPROVED** [review]
Like that you also addressed Android part of requesting permissions for getting BSSID

**Comment by miquelbeltran** [comment]
Thanks! Yeah, since the permission_handler covers both, it was a no-brainer.

**Pull request merged**